doc/sphinx/proof-engine/vernacular-commands.rst

```diff
@@ -1072,53 +942,56 @@ described first.
 
-   This command generalizes the behavior of Opaque and Transparent
+   .. prodn::
+      strategy_level ::= opaque
+      | @int
+      | expand
+      | transparent
+
+   This command accepts the :attr:`local` attribute, which limits its effect
+   to the current section or module, in which case the section and module
+   behavior is the same as :cmd:`Opaque` and :cmd:`Transparent` (without :attr:`global`).
+
+   This command generalizes the behavior of the :cmd:`Opaque` and :cmd:`Transparent`
    commands. It is used to fine-tune the strategy for unfolding
    constants, both at the tactic level and at the kernel level. This
-   command associates a level to the qualified names in the :n:`@qualid`
+   command associates a :n:`@strategy_level` with the qualified names in the :n:`@smart_qualid`
    sequence. Whenever two
    expressions with two distinct head constants are compared (for
    instance, this comparison can be triggered by a type cast), the one
    with lower level is expanded first. In case of a tie, the second one
    (appearing in the cast type) is expanded.
 
-   .. prodn:: level ::= {| opaque | @num | expand }
-
    Levels can be one of the following (higher to lower):
 
     + ``opaque`` : level of opaque constants. They cannot be expanded by
       tactics (behaves like +∞, see next item).
-    + :n:`@num` : levels indexed by an integer. Level 0 corresponds to the
+    + :n:`@int` : levels indexed by an integer. Level 0 corresponds to the
       default behavior, which corresponds to transparent constants. This
       level can also be referred to as transparent. Negative levels
       correspond to constants to be expanded before normal transparent
       constants, while positive levels correspond to constants to be
       expanded after normal transparent constants.
     + ``expand`` : level of constants that should be expanded first (behaves
       like −∞)
+    + ``transparent`` : TBA
 
-    .. cmdv:: Local Strategy @level [ {+ @qualid } ]
-
-       These directives survive section and module closure, unless the
-       command is prefixed by ``Local``. In the latter case, the behavior
-       regarding sections and modules is the same as for the :cmd:`Transparent` and
-       :cmd:`Opaque` commands.
-
-
-.. cmd:: Print Strategy @qualid
+.. cmd:: Print Strategy @smart_qualid
 
-   This command prints the strategy currently associated to :n:`@qualid`. It
-   fails if :n:`@qualid` is not an unfoldable reference, that is, neither a
+   This command prints the strategy currently associated with :n:`@smart_qualid`. It
+   fails if :n:`@smart_qualid` is not an unfoldable reference, that is, neither a
    variable nor a constant.
 
    .. exn:: The reference is not unfoldable.
       :undocumented:
 
-   .. cmdv:: Print Strategies
+.. cmd:: Print Strategies
 
-      Print all the currently non-transparent strategies.
+   Print all the currently non-transparent strategies.
 
 
 .. cmd:: Declare Reduction @ident := @red_expr
```

Comment:
In the description of this command, the `Local` modifier should be replaced by the `local` attribute for consistency with the rest of the documentation.

Reply:
"For the moment, the name is not qualified." Is it or isn't it?

Seems like this may be another obscure feature that's not so valuable.

```
   Declares a short name for the reduction expression :n:`@red_expr`, for
   instance ``lazy beta delta [foo bar]``. This short name can then be used
   in :n:`Eval @ident in` or ``eval`` constructs. This command
   accepts the :attr:`local` attribute, which indicates that the reduction
   will be discarded at the end of the
```

Reply:
No, that's quite valuable I think. Anyway, I know at least Jason Gross is a user. This is useful if you want to use `Eval ... in` with a custom reduction strategy without having to repeat the (possibly complex) reduction strategy each time. However, there is no justification for this name being not qualified. That's also the case of a bunch of other things: scopes, hint databases, etc. They should all become qualified, but this might create compatibility issues.

Reply:
I removed "For the moment".

Is it worth creating an enhancement request to use more qualified identifiers in more places?

Reply:
Yes!

Reply:
See #12063.